### ISSUE TITLE
How to solve: 'Error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class'?

### ISSUE THREAD

**Issue by MarGer1962** [FirebaseExtended/codelab-kanban-fire#22]
Hi,
When i complete making the app, and give the command 'ng serve' i get these messages:

Error: node_modules/@angular/material/form-field/form-field.d.ts:64:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

This likely means that the library (@angular/material/form-field) which declares MatFormField has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to be compatible with Ivy.

64 export declare class MatFormField extends _MatFormFieldMixinBase implements AfterContentInit, AfterContentChecked, AfterViewInit, OnDestroy, CanColor {
                        ~~~~~~~~~~~~
Error: node_modules/@angular/material/form-field/label.d.ts:10:22 - error NG6002: Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

This likely means that the library (@angular/material/form-field) which declares MatLabel has not been processed correctly by ngcc, or is not compatible with Angular Ivy. Check if a newer version of the library is available, and update if so. Also consider checking with the library's authors to see if the library is expected to 
be compatible with Ivy.

10 export declare class MatLabel {
                        ~~~~~~~~

I tried the solution mentioned in 
https://lifesaver.codes/answer/ivy-error-ng6002-appears-in-the-ngmodule-imports-of-appmodule-but-could-not-be-resolved-to-an-ngmodule-class-35399

But that does not work.
Has some one any ideas?
I also tried the solution -'Check if a newer version of the library is available'-  but that caused a lot of other problems, so many that i had to rebuild using the package.json file!
But that does not work as well.

Would be great, this is driving me crazy!

**Comment by MarGer1962** [comment]
LS,
I tried to make the app working by following all the steps in the tutorial. I could not get it working - that is: after the part 8 (connecting to a database).
So : i updated the packagefiles > it would not work. 
Then tried the solution > see: naming is way off. #20 

Then i started a new project while using the package.json file included in here (https://github.com/FirebaseExtended/codelab-kanban-fire). Copied the content of all the src files of the project.
Does not seem to work either. I do not know what i am doing wrong. I feel kinda stupid.

**Comment by rutech-rmi** [comment]
With the switch to Firebase, the Task[] type of todo, inProgress, and done is now wrapped into an Observable as declared in `app.component.ts`

> `this.store.collection('todo').valueChanges({ idField: 'id' }) as Observable<Task[]>;`

Angular subscribes to an Observable automatically, by adding the async pipe ` | async` in `app.component.html`

- Change `[cdkDropListData]="todo"` to `[cdkDropListData]="todo | async"`

This also changes the parameter type that is passed to the drop() function. You may add `any` to make your code work: 
`drop(event: CdkDragDrop<Task[] | any>): void`

Also, you will need to add an type import for the Observable
`import {Observable } from 'rxjs';`

The next step has a [final solution](https://github.com/FirebaseExtended/codelab-kanban-fire/blob/workshop/src/app/app.component.ts), but this one makes it work for step 11 [Moving the data to Firestore](https://developers.google.com/codelabs/building-a-web-app-with-angular-and-firebase#10).

**Comment by MarGer1962** [comment]
Thanks, 
I analysed my code after your reply  and saw that i mixed up `import {Observable } from 'rxjs';` and `import { BehaviorSubject } from 'rxjs';` in the `app.component.ts`
I repaired that and it's is working! So problem solved.

Regarding my problem with Angular Material (MatLabel and MatFormField). I solved this.
Solution: 
in `app.module.ts` i replaced `import { MatFormField } from '@angular/material/form-field';` 
with this line of code `{ MatFormFieldModule } from '@angular/material/form-field'; `
and put the `MatFormFieldModule` in `@NgModule`, like so

```
@NgModule({
imports:[
...
MatFormFieldModule,
...

```
